### PR TITLE
Add support for windows path uris

### PR DESCRIPTION
**Pull request by akermu**
https://en.wikipedia.org/wiki/File_URI_scheme#Windows_2

### CODE REVIEW

File: lsp-common.el [hunk 1]
```diff
@@ -28,6 +28,11 @@
   :group 'lsp-mode
   :type 'boolean)
 
+(defvar lsp--uri-file-prefix (pcase system-type
```
Comment (vibhavp) [inline lsp-common.el]: Wouldn't using `eval-when-compile` be a better idea here?

Comment (akermu) [reply lsp-common.el]: Compiled elisp files should be portable across operating systems and a `eval-when-compile` in this case would violate this condition.

Comment (vibhavp) [reply lsp-common.el]: ah, okay

**Comment by vibhavp** [comment]
Looks good, thanks.

**Pull request merged**